pyccel/ast/builtin_methods/set_methods.py

```diff
@@ -104,3 +107,42 @@ class SetClear(SetMethod):
 
     def __init__(self, set_variable):
         super().__init__(set_variable)
+
+
+class SetRemove(SetMethod):
+    """
+    Represents a call to the .remove() method.
+
+    The remove() removes the specified item from 
+    the set and updates the set. It doesn't return any value.
+
+    Parameters
+    ----------
+    set_variable : TypedAstNode
+        The name of the set.
+
+    item : TypedAstNode
+        The item to search for, and remove.
+    """
+    __slots__ = ()
+    _dtype = NativeVoid()
+    _shape = None
+    _order = None
+    _rank = 0
+    _precision = None
+    _class_type = NativeVoid()
+    name = 'remove'
+
+    def __init__(self, set_variable, item) -> None:
+        if (isinstance(item, PythonRange)):
+            raise TypeError("KeyError : ")
```

Comment:
I am surprised by this code. Why is a range a special case? Are there no other objects that fail (e.g. enumerate)? The error message is also not very verbose. The user will see:
```
Error : Key Error
```
And their code. This may not be very clear

Reply:
Should the error message be formatted like Python?

Reply:
No. It cannot be formatted like Python as Python's error says "I could not find the object that you are looking for in this set". Your error says something different. It says "It is not possible to look for a {item} in a set of {elem_type}". This is not equivalent

Reply:
Fixed : d2a8a8f

Reply:
Why not check the dtype after you have checked that the type matches the documented type?

Reply:
That seems logical given that it is not a `TypedAstNode` and your docstring states that you only support `TypedAstNode`s

Reply:
you're right, What should I update the docstring with?

Reply:
All you are doing here is checking that the docstring is respected

Reply:
The docstring describes the assumptions for your function. You can either raise an error if the arguments don't respect these assumptions or leave it up to the user to ensure that they never call the function with arguments of the wrong type

Reply:
Oh, got it! I see your point now.